pkg/services/store/object/http.go

```diff
@@ -0,0 +1,165 @@
+	uid, kind, params := parseRequestParams(c.Req)
+	rsp, err := s.store.Read(c.Req.Context(), &ReadObjectRequest{
+		UID:         uid,
+		Kind:        kind,
+		Version:     params["version"], // ?version = XYZ
+		WithBody:    true,              // ?? allow false?
+		WithSummary: true,              // ?? allow false?
+	})
+	if err != nil {
+		return response.Error(500, "?", err)
+	}
+	if rsp.Object == nil {
+		rsp.Object = &RawObject{
+			UID:      uid,
+			Kind:     "missing!",
+			Modified: time.Now().UnixMilli(),
+		}
+	}
+	return response.JSON(200, rsp)
+}
+
+func (s *httpObjectStore) doGetRawOject(c *models.ReqContext) response.Response {
+	uid, kind, params := parseRequestParams(c.Req)
+	rsp, err := s.store.Read(c.Req.Context(), &ReadObjectRequest{
+		UID:         uid,
+		Kind:        kind,
+		Version:     params["version"], // ?version = XYZ
+		WithBody:    true,
+		WithSummary: false,
+	})
+	if err != nil {
+		return response.Error(500, "?", err)
+	}
+	if rsp.Object != nil && rsp.Object.Body != nil {
+		return response.CreateNormalResponse(
+			http.Header{
+				"Content-Type": []string{"application/json"}, // TODO, based on kind!!!
+			},
+			rsp.Object.Body,
+			200,
+		)
+	}
+	return response.JSON(400, rsp) // ???
+}
+
+func (s *httpObjectStore) doWriteObject(c *models.ReqContext) response.Response {
+	uid, kind, params := parseRequestParams(c.Req)
+	b, err := ioutil.ReadAll(c.Req.Body)
+	if err != nil {
+		return response.Error(400, "error reading body", err)
+	}
+
+	rsp, err := s.store.Write(c.Req.Context(), &WriteObjectRequest{
+		UID:             uid,
+		Kind:            kind,
+		Body:            b,
+		Comment:         params["comment"],
+		PreviousVersion: params["previous"],
+	})
+	if err != nil {
+		return response.Error(500, "?", err)
+	}
+	return response.JSON(200, rsp)
+}
+
+func (s *httpObjectStore) doDeleteObject(c *models.ReqContext) response.Response {
+	uid, kind, params := parseRequestParams(c.Req)
+	rsp, err := s.store.Delete(c.Req.Context(), &DeleteObjectRequest{
+		UID:             uid,
+		Kind:            kind,
+		PreviousVersion: params["previous"],
+	})
+	if err != nil {
+		return response.Error(500, "?", err)
+	}
+	return response.JSON(200, rsp)
+}
+
+func (s *httpObjectStore) doGetHistory(c *models.ReqContext) response.Response {
+	uid, kind, params := parseRequestParams(c.Req)
+	limit := int64(20) // params
+	rsp, err := s.store.History(c.Req.Context(), &ObjectHistoryRequest{
+		UID:           uid,
+		Kind:          kind,
+		Limit:         limit,
+		NextPageToken: params["nextPageToken"],
+	})
+	if err != nil {
+		return response.Error(500, "?", err)
+	}
+	return response.JSON(200, rsp)
+}
+
+func (s *httpObjectStore) doListFolder(c *models.ReqContext) response.Response {
+	return response.JSON(500, "Not implemented yet")
+}
+
+func (s *httpObjectStore) doSearch(c *models.ReqContext) response.Response {
+	return response.JSON(500, "Not implemented yet")
```

Comment:
Minor but for these we can use [http code 501](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501) for not implemented yet :)